lua/toggleterm.lua

```diff
@@ -35,19 +35,29 @@ end
 
 local function setup_global_mappings()
   local mapping = config.open_mapping
-  -- v:count defaults the count to 0 but if a count is passed in uses that instead
-  if mapping then
-    vim.keymap.set("n", mapping, '<Cmd>execute v:count . "ToggleTerm"<CR>', {
+
+  -- Key mapping function
+  ---@param key string
+  local map = function(key)
+    -- v:count defaults the count to 0 but if a count is passed in uses that instead
+    vim.keymap.set("n", key, '<Cmd>execute v:count . "ToggleTerm"<CR>', {
       desc = "Toggle Terminal",
       silent = true,
     })
     if config.insert_mappings then
-      vim.keymap.set("i", mapping, "<Esc><Cmd>ToggleTerm<CR>", {
+      vim.keymap.set("i", key, "<Esc><Cmd>ToggleTerm<CR>", {
         desc = "Toggle Terminal",
         silent = true,
       })
     end
   end
+
+  if type(mapping) == "string" then map(mapping) end
```

Comment:
Although it's impossible (🤞🏾) I'd rather this was an `if then else if` I feel uncomfortable with the idea that it could possible fall into the next clause even though it can't